### PR TITLE
Update to use `node16` module resolution in `tsconfig.json`

### PR DESCRIPTION
**Pull request by ChristianMurphy**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

Updates TypeScript configuration to use node16 resolution for ESM.

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
Depends on https://github.com/remarkablemark/style-to-object/pull/47

**Comment by remarkablemark** [comment]
> Depends on [remarkablemark/style-to-object#47](https://github.com/remarkablemark/style-to-object/pull/47)

@ChristianMurphy `style-to-object@0.4.1` has been released

**Comment by ChristianMurphy** [comment]
Thanks @remarkablemark! 🙇 

@remarkjs/maintainers this is now ready for a review.

File: tsconfig.json [hunk 1]
```diff
@@ -2,19 +2,15 @@
...
   "compilerOptions": {
-    "target": "ES2020",
-    "lib": ["ES2020", "DOM"],
-    "module": "ES2020",
-    "moduleResolution": "node",
+    "target": "es2021",
+    "lib": ["es2020", "dom"],
+    "module": "node16",
     "jsx": "react",
     "allowJs": true,
```
Comment (remcohaszing) [inline tsconfig.json:9]: Same nit as before:

```suggestion
```

Looks great otherwise! :+1:

Comment (ChristianMurphy) [reply tsconfig.json]: Thanks for the reminder!
Updated

**Review by remcohaszing — APPROVED** [review]

**Review by wooorm — APPROVED** [review]

**Pull request merged**